lib/datadog/core/remote/component.rb

```diff
@@ -20,7 +20,20 @@ def initialize(settings, agent_settings)
           transport_v7 = Datadog::Core::Transport::HTTP.v7(**transport_options.dup)
 
           @client = Client.new(transport_v7)
-          @worker = Worker.new(interval: settings.remote.poll_interval_seconds) { @client.sync }
+          @worker = Worker.new(interval: settings.remote.poll_interval_seconds) do
+            begin
+              @client.sync
+            rescue StandardError => e
+              Datadog.logger.error do
+                "remote worker error: #{e.class.name} #{e.message} location: #{Array(e.backtrace).first}"
+              end
+
+              # client state is unknown, state might be corrupted
+              @client = Client.new(transport_v7)
```

Comment:
Why don't we add a method on the client that resets the internal state? 

`@client.reset` will make sure repository is a branch new after calling `reset`

Reply:
Such resets should be avoided at all costs as they provide a dangerous public API, especially in concurrent contexts.

In this case it would mean:
- changing the client id, which conceptually makes no sense: the instance immediately has an id for its whole lifetime
- it would require resetting the repository and any other internal state

There is already a facility for that: `new` creates an exactly new client.

IOW it would change the class from encapsulating the notion of being a specific client to being a global singleton in disguise.

Reply:
Sidenote: it also points to the client's dispatcher having receivers be registered to not be encapsulated in the client but done by the component, something I anticipated to be the case even before this change.